README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

# Animated Card Flipper Screen

**Card Flipper Screen** package lets you add an animated card flipper dashboard to your Flutter app.

## Features

The CardFlipperScreen widget is built to be `a show model screen` to any content in your Flutter app. By using the `cardsToDispaly` property, you can provide the content to be displayed.

_The package will handle the animation by itself._

https://user-images.githubusercontent.com/68671238/165011475-ce673de8-72c3-47c9-a863-eca2dcf977c7.mp4

<hr>

## Getting started

1. Add the latest version of package to your `pubspec.yaml` (and run `dart pub get`):

```dart
dependencies:
  card_flipper: ^1.0.2
```

2. Import the package and use it in your Flutter App.

```dart
import 'package:card_flipper/card_flipper.dart';
```

<hr>

## Usage

There are a number of properties that you can modify:

- cardsToDispaly (cards content)
- topSpace (top page space)
- backgroundColor
- backWidget
- optionalActionWidget
- onTapBackBtn
- onTapOptionalActionWidget
- onTapFunction
- topSectionTitleStyle
- middleSectionTitleStyle
- middleSectionDescriptionStyle
- mainBtnWidget

**Example Usage ( complete with all params ):**

<table>
 <tr>
 <td>
      
```dart
class HomePage extends StatefulWidget {
  const HomePage({Key? key}) : super(key: key);

@override
State<HomePage> createState() => \_HomePageState();
}

class _HomePageState extends State<HomePage> {
  final List<CardViewModel> _cardsToDisplay = [
    CardViewModel(
        backdropAssetPath: 'assets/images/china.jpg',
        address: 'China',
        capitol: 'Beijing',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/korea.jpg',
        address: 'Korea',
        capitol: 'Seoul',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/germany.jpg',
        address: 'Germany',
        capitol: 'Berlin',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/italy.jpg',
        address: 'Italy',
        capitol: 'Rome',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/switzerland.jpg',
        address: 'Switzerland',
        capitol: 'Bern',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CardFlipperScreen(
        cardsToDispaly: _cardsToDisplay,
        topSpace: 20.0,
        backgroundColor: Colors.black,
        backWidget: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: const [
            Icon(
              Icons.arrow_back_ios,
              color: Colors.white,
              size: 18.0,
            ),
            SizedBox(
              width: 6.0,
            ),
            Text(
              'back',
              style: TextStyle(
                fontFamily: 'Montserrat',
                fontSize: 14.0,
                fontWeight: FontWeight.bold,
                letterSpacing: 2.0,
                color: Colors.white,
              ),
            ),
          ],
        ),
        optionalActionWidget: const Icon(
          Icons.bookmarks_outlined,
          color: Colors.white,
        ),
        onTapBackBtn: () {},
        onTapOptionalActionWidget: () {},
        onTapFunction: () {},
        topSectionTitleStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 20.0,
          fontWeight: FontWeight.bold,
          letterSpacing: 2.0,
          color: Colors.white,
        ),
        middleSectionTitleStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 80.0,
          fontWeight: FontWeight.bold,
          letterSpacing: -4.0,
          color: Colors.white,
        ),
        middleSectionDescriptionStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 16.0,
          fontWeight: FontWeight.normal,
          letterSpacing: 1.0,
          color: Colors.white,
        ),
        mainBtnWidget: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          mainAxisSize: MainAxisSize.min,
          children: const [
            Text(
              'Explore',
              style: TextStyle(
                fontFamily: 'Montserrat',
                fontSize: 16.0,
                fontWeight: FontWeight.bold,
                letterSpacing: 2.0,
                color: Colors.white,
              ),
            ),
            Padding(
              padding: EdgeInsets.only(
                left: 10.0,
              ),
              child: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
                size: 16.0,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```
   </td>
   <td>
     Here's what it looks like:

<hr>


https://user-images.githubusercontent.com/68671238/165011475-ce673de8-72c3-47c9-a863-eca2dcf977c7.mp4


   </td>
  </tr>
  </table>
<hr>

## Next Goals
We are working on some improvements including:

- [ ] Make more properties customizable.
- [ ] Make the cards animation more cool looking.

## Issues & Feedback
Please file an [issue!](https://github.com/aliMissaoui/Card-Flipper-Package/issues) to send feedback or report a bug. Thank you!

```


CHANGELOG.md
## 1.0.0

- Animated Card Flipper Dashboard.

## 1.0.1

- Update README file : better docs.

## 1.0.2

- Update package.


lib/src/widgets/card.dart
import 'package:card_flipper/src/model/card_data_model.dart';
import 'package:flutter/material.dart';

class CardWidget extends StatelessWidget {
  final CardViewModel viewModel;
  final double parallaxPercent;
  final TextStyle? topSectionTitleStyle;
  final TextStyle? middleSectionTitleStyle;
  final TextStyle? middleSectionDescriptionStyle;
  final Widget? mainBtnWidget;
  final VoidCallback? onTapFunction;
  const CardWidget(
      {Key? key,
      required this.viewModel,
      this.parallaxPercent = 0.0,
      this.topSectionTitleStyle,
      this.middleSectionTitleStyle,
      this.middleSectionDescriptionStyle,
      this.mainBtnWidget, this.onTapFunction})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Stack(
      fit: StackFit.expand,
      children: [
        // Background
        ClipRRect(
          borderRadius: BorderRadius.circular(10.0),
          child: FractionalTranslation(
            translation: Offset(parallaxPercent * 4.0, 0.0),
            child: OverflowBox(
              maxWidth: double.infinity,
              child: Image.asset(
                viewModel.backdropAssetPath,
                fit: BoxFit.cover,
              ),
            ),
          ),
        ),
        Container(
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(10.0),
            // color: Colors.black.withOpacity(0.6),
            gradient: LinearGradient(
              begin: Alignment.topRight,
              end: Alignment.bottomLeft,
              colors: [
                Colors.black.withOpacity(0.1),
                Colors.black.withOpacity(0.6)
              ],
            ),
          ),
        ),
        // Content
        Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: [
            Padding(
              padding: const EdgeInsets.only(
                top: 30.0,
                left: 20.0,
                right: 20.0,
              ),
              child: Text(
                viewModel.address.toUpperCase(),
                style: topSectionTitleStyle ??
                    const TextStyle(
                      fontFamily: 'Montserrat',
                      fontSize: 20.0,
                      fontWeight: FontWeight.bold,
                      letterSpacing: 2.0,
                      color: Colors.white,
                    ),
              ),
            ),
            Expanded(child: Container()),
            Padding(
              padding: const EdgeInsets.only(
                left: 20.0,
                right: 20.0,
              ),
              child: Text(
                viewModel.capitol,
                style: middleSectionTitleStyle ??
                    const TextStyle(
                      fontFamily: 'Montserrat',
                      fontSize: 80.0,
                      fontWeight: FontWeight.bold,
                      letterSpacing: -4.0,
                      color: Colors.white,
                    ),
              ),
            ),
            Padding(
              padding: const EdgeInsets.only(
                left: 20.0,
                right: 20.0,
              ),
              child: Text(
                viewModel.description,
                style: middleSectionDescriptionStyle ??
                    const TextStyle(
                      fontFamily: 'Montserrat',
                      fontSize: 16.0,
                      fontWeight: FontWeight.normal,
                      letterSpacing: 1.0,
                      color: Colors.white,
                    ),
                maxLines: 3,
                textAlign: TextAlign.center,
                softWrap: false,
                overflow: TextOverflow.ellipsis,
              ),
            ),
            Expanded(child: Container()),
            Padding(
              padding: const EdgeInsets.only(
                top: 50.0,
                bottom: 50.0,
              ),
              child: Material(
                color: Colors.transparent,
                borderRadius: BorderRadius.circular(30.0),
                child: InkWell(
                  borderRadius: BorderRadius.circular(30.0),
                  onTap: onTapFunction ?? () {},
                  child: Container(
                    decoration: BoxDecoration(
                      borderRadius: BorderRadius.circular(30.0),
                      border: Border.all(
                        color: Colors.white,
                        width: 1.5,
                      ),
                      color: Colors.black.withOpacity(0.3),
                    ),
                    child: Padding(
                      padding: const EdgeInsets.symmetric(
                        vertical: 8.0,
                        horizontal: 24.0,
                      ),
                      child: mainBtnWidget ??
                          Row(
                            mainAxisAlignment: MainAxisAlignment.center,
                            mainAxisSize: MainAxisSize.min,
                            children: const [
                              Text(
                                'Explore',
                                style: TextStyle(
                                  fontFamily: 'Montserrat',
                                  fontSize: 16.0,
                                  fontWeight: FontWeight.bold,
                                  letterSpacing: 2.0,
                                  color: Colors.white,
                                ),
                              ),
                              Padding(
                                padding: EdgeInsets.only(
                                  left: 10.0,
                                ),
                                child: Icon(
                                  Icons.arrow_forward_ios,
                                  color: Colors.white,
                                  size: 16.0,
                                ),
                              ),
                            ],
                          ),
                    ),
                  ),
                ),
              ),
            ),
          ],
        ),
      ],
    );
  }
}


lib/src/card_flipper_page.dart
import 'package:card_flipper/src/model/card_data_model.dart';
import 'package:card_flipper/src/widgets/bottombar.dart';
import 'package:card_flipper/src/widgets/card_flipper.dart';
import 'package:flutter/material.dart';

class CardFlipperScreen extends StatefulWidget {
  ///Set App Name
  final double topSpace;

  /// Set Cards to display as Card View Models
  final List<CardViewModel> cardsToDispaly;

  /// Set the back button widget
  final Widget? backWidget;

  /// Set the optional widget
  final Widget? optionalActionWidget;

  /// Handle click back widget
  final VoidCallback? onTapBackBtn;

  /// Handle click optional widget
  final VoidCallback? onTapOptionalActionWidget;

  /// Set top section title style
  final TextStyle? topSectionTitleStyle;

  /// Set middle section title style
  final TextStyle? middleSectionTitleStyle;

  /// Set middle section decription style
  final TextStyle? middleSectionDescriptionStyle;

  /// Set main button widget
  final Widget? mainBtnWidget;

  /// Set on tap function on the main button
  final VoidCallback? onTapFunction;

  //Set background color
  final Color? backgroundColor;

  const CardFlipperScreen(
      {Key? key,
      this.topSpace = 20.0,
      required this.cardsToDispaly,
      this.backWidget,
      this.optionalActionWidget,
      this.onTapBackBtn,
      this.onTapOptionalActionWidget,
      this.topSectionTitleStyle,
      this.middleSectionTitleStyle,
      this.middleSectionDescriptionStyle,
      this.mainBtnWidget,
      this.onTapFunction, this.backgroundColor})
      : super(key: key);

  @override
  State<CardFlipperScreen> createState() => _CardFlipperScreenState();
}

class _CardFlipperScreenState extends State<CardFlipperScreen> {
  double scrollPercent = 0.0;
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: widget.backgroundColor ?? Colors.black,
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          SizedBox(
            width: double.infinity,
            height: widget.topSpace,
          ),
          Expanded(
            child: CardFlipperWidget(
              topSectionTitleStyle: widget.topSectionTitleStyle,
              middleSectionTitleStyle: widget.middleSectionTitleStyle,
              middleSectionDescriptionStyle:
                  widget.middleSectionDescriptionStyle,
              mainBtnWidget: widget.mainBtnWidget,
              onTapFunction: widget.onTapFunction,
              cards: widget.cardsToDispaly,
              onScroll: (double scrollPercent) {
                setState(() {
                  this.scrollPercent = scrollPercent;
                });
              },
            ),
          ),
          BottomBar(
            backWidget: widget.backWidget,
            optionalActionWidget: widget.optionalActionWidget,
            onTapBackBtn: widget.onTapBackBtn,
            onTapOptionalActionWidget: widget.onTapOptionalActionWidget,
            scrollPercent: scrollPercent,
            cardCount: widget.cardsToDispaly.length,
          ),
        ],
      ),
    );
  }
}


lib/src/widgets/bottombar.dart
import 'package:card_flipper/src/widgets/scroll_indicator.dart';
import 'package:flutter/material.dart';

class BottomBar extends StatelessWidget {
  final int cardCount;
  final double scrollPercent;
  final Widget? backWidget;
  final Widget? optionalActionWidget;
  final VoidCallback? onTapBackBtn;
  final VoidCallback? onTapOptionalActionWidget;
    final Color? trackPaintColor;
  final Color? thumbPaintColor;


  const BottomBar(
      {Key? key, required this.cardCount, required this.scrollPercent, this.backWidget, this.onTapBackBtn, this.optionalActionWidget, this.onTapOptionalActionWidget, this.trackPaintColor, this.thumbPaintColor})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(top: 15.0, bottom: 15.0),
      child: Row(
        children: [
          Expanded(
            child: Center(
              child: GestureDetector(
                onTap: onTapBackBtn ?? () => Navigator.pop(context),
                child: backWidget ?? Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: const [
                    Icon(
                      Icons.arrow_back_ios,
                      color: Colors.white,
                      size: 18.0,
                    ),
                    SizedBox(
                      width: 6.0,
                    ),
                    Text(
                      'back',
                      style: TextStyle(
                        fontFamily: 'Montserrat',
                        fontSize: 14.0,
                        fontWeight: FontWeight.bold,
                        letterSpacing: 2.0,
                        color: Colors.white,
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ),
          Expanded(
              child: SizedBox(
            width: double.infinity,
            height: 5.0,
            child: ScrollIndicator(
              trackPaintColor: trackPaintColor,
              thumbPaintColor: thumbPaintColor,
              cardCount: cardCount,
              scrollPercent: scrollPercent,
            ),
          )),
          Expanded(
            child: Center(
              child: GestureDetector(
                onTap: onTapOptionalActionWidget ?? () {},
                child: optionalActionWidget ?? const Icon(
                  Icons.bookmarks_outlined,
                  color: Colors.white,
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}


lib/src/widgets/card_flipper.dart
import 'dart:ui';

import 'package:card_flipper/src/model/card_data_model.dart';
import 'package:card_flipper/src/widgets/card.dart';
import 'package:flutter/material.dart';

class CardFlipperWidget extends StatefulWidget {
  final List<CardViewModel> cards;
  final Function(double scrollPercent) onScroll;
    final TextStyle? topSectionTitleStyle;
  final TextStyle? middleSectionTitleStyle;
  final TextStyle? middleSectionDescriptionStyle;
  final Widget? mainBtnWidget;
  final VoidCallback? onTapFunction;

  const CardFlipperWidget(
      {Key? key, required this.cards, required this.onScroll, this.topSectionTitleStyle, this.middleSectionTitleStyle, this.middleSectionDescriptionStyle, this.mainBtnWidget, this.onTapFunction})
      : super(key: key);

  @override
  State<CardFlipperWidget> createState() => _CardFlipperWidgetState();
}

class _CardFlipperWidgetState extends State<CardFlipperWidget>
    with TickerProviderStateMixin {
  var scrollPercent = 0.0;
  late Offset startDrag;
  late double startDragPercentScroll;
  late double finishScrollStart;
  late double finishScrollEnd;
  late AnimationController finishScrollController;

  @override
  void initState() {
    super.initState();

    finishScrollController = AnimationController(
        duration: const Duration(milliseconds: 150), vsync: this)
      ..addListener(() {
        setState(() {
          scrollPercent = lerpDouble(finishScrollStart, finishScrollEnd,
              finishScrollController.value)!;

      widget.onScroll(scrollPercent);

        });
      });
  }

  @override
  void dispose() {
    finishScrollController.dispose();
    super.dispose();
  }

  void _onHorizontalDragStart(DragStartDetails details) {
    startDrag = details.globalPosition;
    startDragPercentScroll = scrollPercent;
  }

  void _onHorizontalDragUpdate(DragUpdateDetails details) {
    final currentDrag = details.globalPosition;
    final dragDistance = currentDrag.dx - startDrag.dx;
    final singleCardDragPercent = dragDistance / context.size!.width;

    setState(() {
      scrollPercent = (startDragPercentScroll +
              (-singleCardDragPercent / widget.cards.length))
          .clamp(0.0, 1.0 - (1 / widget.cards.length));

      widget.onScroll(scrollPercent);
    });
  }

  void _onHorizontalDragEnd(DragEndDetails details) {
    finishScrollStart = scrollPercent;
    finishScrollEnd =
        (scrollPercent * widget.cards.length).round() / widget.cards.length;
    finishScrollController.forward(from: 0.0);

    setState(() {
      startDrag = const Offset(0.0, 0.0);
      startDragPercentScroll = 0.0;
    });
  }

  List<Widget> _buildCards() {
    final cardsCount = widget.cards.length;

    int index = -1;
    return widget.cards.map<Widget>((CardViewModel cardViewModel) {
      ++index;
      return _buildCard(widget.cards[index], index, cardsCount, scrollPercent);
    }).toList();
  }

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onHorizontalDragStart: _onHorizontalDragStart,
      onHorizontalDragUpdate: _onHorizontalDragUpdate,
      onHorizontalDragEnd: _onHorizontalDragEnd,
      behavior: HitTestBehavior.translucent,
      child: Stack(
        children: _buildCards(),
      ),
    );
  }

  _buildCard(CardViewModel viewModel, int cardIndex, int cardCount,
      double scrollPercent) {
    final cardScrollPercent = scrollPercent / (1 / cardCount);
    final parallax = scrollPercent - (cardIndex / cardCount);

    return FractionalTranslation(
        translation: Offset(cardIndex - cardScrollPercent, 0.0),
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: CardWidget(
            mainBtnWidget: widget.mainBtnWidget,
            onTapFunction: widget.onTapFunction,
            topSectionTitleStyle: widget.topSectionTitleStyle,
            middleSectionDescriptionStyle: widget.middleSectionDescriptionStyle,
            middleSectionTitleStyle: widget.middleSectionDescriptionStyle,
            viewModel: viewModel,
            parallaxPercent: parallax,
          ),
        ));
  }
}


lib/src/widgets/scroll_indicator.dart
import 'package:flutter/material.dart';

class ScrollIndicator extends StatelessWidget {
  final int cardCount;
  final double scrollPercent;
  final Color? trackPaintColor;
  final Color? thumbPaintColor;

  const ScrollIndicator(
      {Key? key, required this.cardCount, required this.scrollPercent, this.trackPaintColor, this.thumbPaintColor})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return CustomPaint(
      painter: ScrollIndicatorPainter(
        cardCount: cardCount,
        scrollPercent: scrollPercent,
      ),
      child: Container(),
    );
  }
}

class ScrollIndicatorPainter extends CustomPainter {
  final int cardCount;
  final double scrollPercent;
  final Paint trackPaint;
  final Paint thumbPaint;
    final Color? trackPaintColor;
  final Color? thumbPaintColor;

  ScrollIndicatorPainter({this.trackPaintColor, this.thumbPaintColor, required this.cardCount, required this.scrollPercent})
      : trackPaint = Paint()
          ..color = trackPaintColor ??const Color(0xff444444)
          ..style = PaintingStyle.fill,
        thumbPaint = Paint()
          ..color = thumbPaintColor ?? Colors.white
          ..style = PaintingStyle.fill;

  @override
  void paint(Canvas canvas, Size size) {
    canvas.drawRRect(
        RRect.fromRectAndCorners(
          Rect.fromLTWH(0.0, 0.0, size.width, size.height),
          topLeft: const Radius.circular(3.0),
          topRight: const Radius.circular(3.0),
          bottomLeft: const Radius.circular(3.0),
          bottomRight: const Radius.circular(3.0),
        ),
        trackPaint);

    final thumbWidth = size.width / cardCount;
    final thumbLeft = scrollPercent * size.width;

    canvas.drawRRect(
        RRect.fromRectAndCorners(
          Rect.fromLTWH(thumbLeft, 0.0, thumbWidth, size.height),
          topLeft: const Radius.circular(3.0),
          topRight: const Radius.circular(3.0),
          bottomLeft: const Radius.circular(3.0),
          bottomRight: const Radius.circular(3.0),
        ),
        thumbPaint);
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) {
    return true;
  }
}


example/lib/main.dart
import 'package:card_flipper/card_flipper.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  const HomePage({Key? key}) : super(key: key);

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  final List<CardViewModel> _cardsToDisplay = [
    CardViewModel(
        backdropAssetPath: 'assets/images/china.jpg',
        address: 'China',
        capitol: 'Beijing',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/korea.jpg',
        address: 'Korea',
        capitol: 'Seoul',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/germany.jpg',
        address: 'Germany',
        capitol: 'Berlin',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/italy.jpg',
        address: 'Italy',
        capitol: 'Rome',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
    CardViewModel(
        backdropAssetPath: 'assets/images/switzerland.jpg',
        address: 'Switzerland',
        capitol: 'Bern',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque lectus molestie faucibus mi. Lorem a ipsum justo, eros in massa justo. '),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CardFlipperScreen(
        cardsToDispaly: _cardsToDisplay,
        topSpace: 20.0,
        backgroundColor: Colors.black,
        backWidget: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: const [
            Icon(
              Icons.arrow_back_ios,
              color: Colors.white,
              size: 18.0,
            ),
            SizedBox(
              width: 6.0,
            ),
            Text(
              'back',
              style: TextStyle(
                fontFamily: 'Montserrat',
                fontSize: 14.0,
                fontWeight: FontWeight.bold,
                letterSpacing: 2.0,
                color: Colors.white,
              ),
            ),
          ],
        ),
        optionalActionWidget: const Icon(
          Icons.bookmarks_outlined,
          color: Colors.white,
        ),
        onTapBackBtn: () {},
        onTapOptionalActionWidget: () {},
        onTapFunction: () {},
        topSectionTitleStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 20.0,
          fontWeight: FontWeight.bold,
          letterSpacing: 2.0,
          color: Colors.white,
        ),
        middleSectionTitleStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 80.0,
          fontWeight: FontWeight.bold,
          letterSpacing: -4.0,
          color: Colors.white,
        ),
        middleSectionDescriptionStyle: const TextStyle(
          fontFamily: 'Montserrat',
          fontSize: 16.0,
          fontWeight: FontWeight.normal,
          letterSpacing: 1.0,
          color: Colors.white,
        ),
        mainBtnWidget: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          mainAxisSize: MainAxisSize.min,
          children: const [
            Text(
              'Explore',
              style: TextStyle(
                fontFamily: 'Montserrat',
                fontSize: 16.0,
                fontWeight: FontWeight.bold,
                letterSpacing: 2.0,
                color: Colors.white,
              ),
            ),
            Padding(
              padding: EdgeInsets.only(
                left: 10.0,
              ),
              child: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
                size: 16.0,
              ),
            ),
          ],
        ),
      ),
    );
  }
}
